src/transformers/utils/chat_parsing_utils.py

```diff
@@ -0,0 +1,236 @@
+from __future__ import annotations
+
+import json
+import re
+
+from transformers.utils import is_jmespath_available
+
+
+if is_jmespath_available():
+    import jmespath
+else:
+    jmespath = None
+
+
+def _parse_re_match(node_match):
```

Comment:
```suggestion
def _parse_re_match(node_match: re.Match) -> dict | str:
```

type hints makes mental parsing simpler :D

(I think the `else` returns a `str`, right?)